### DISCUSSION TITLE
Metadata API only available in server side rendering

### DISCUSSION THREAD

**Discussion by hskang9** [vercel/next.js#64897] [App Router]
Hey, I am trying to use metadata in a page while using useState. Using Head component made me to get the data from the page in client and change it. However, using metadata only works when a page is server side rendered. Is there any workaround in this? Current migration experience is just awful.

**Comment by SouthLink** [comment]
You don't have to change information like the page title using data in the metadata from the client component, because that would work for the user, but it might not work for google bot.

You can use dynamic-metadata to dynamically change the SEO information of the page in the server component:

https://nextjs.org/docs/app/building-your-application/optimizing/metadata#dynamic-metadata
